### ISSUE TITLE
[HOLD for payment 2023-07-26] [$1000] Web -  The tooltip end is broken for users with long names on details page of group chat

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#21100]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to staging dot on web chrome and login with User A
2. Create a group with multiple users
3. Click on details page
4. Login with any one user on another web chrome that you have create a group with
5. Change the display name of the user and replace it with a long name (max character)
6. Go to User A and hover over the username of the user on the details page and notice that the pointer of the tooltip is broken (not centered)

## Expected Result:
The pointer of the tooltip should be centered

## Actual Result:
The pointer of the tooltip is off-center for users with long names on details page of group chat

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.29.2

**Reproducible in staging?:** y

**Reproducible in production?:** y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/787c37be-8135-42c8-a805-283d609f2ea0

https://github.com/Expensify/App/assets/93399543/c3a93d49-3fe4-42aa-9d18-6bb02515b3d3

**Expensify/Expensify Issue URL:**

**Issue reported by:** @priya-zha

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686635011685439

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01813ef7a0589a7c4b</li>
        <li>Upwork Job ID: 1676961079120977920</li>
        <li>Last Price Increase: 2023-07-06</li>
    </ul>
</details>

**Comment by johncschuster** [comment]
Looks like @alitoshmatov has a proposal in the above-linked GH that should fix this as well.

**Comment by johncschuster** [comment]
I'm going to downgrade this to `Weekly` while we wait on the other issue to get resolved.

**Comment by alitoshmatov** [comment]
@johncschuster Looks like the other issue is closed, I think you should proceed with this one. Just in case, I am going post my proposal here as well

**Comment by alitoshmatov** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Web -Tooltip for user in group is offcenter and displayed over report
### What is the root cause of that problem?
We are not passing `shiftHorizontal` prop down to tooltip in `UserDetailsTooltip` which is responsible for preventing this kind of issue, where child element is not visible and its center is outside of parent element. We are passing `shiftHorizontal` to `UserDetailsTooltip`

https://github.com/Expensify/App/blob/ddc19e004211ce4282c2c28b08dafadd84a62cc5/src/components/DisplayNames/index.js#L100

### What changes do you think we should make in order to solve the problem?
We should pass `props.shiftHorizontal` down to `Tooltip` component here:
https://github.com/Expensify/App/blob/e6c2704eb43d4d31a55eff5d5d1b3e89c5292bc3/src/components/UserDetailsTooltip/index.js#L42
```javascript
  shiftHorizontal={props.shiftHorizontal}
``` 

Result:
<img width="1440" alt="Screenshot 2023-06-30 at 12 40 50 PM" src="https://github.com/Expensify/App/assets/59907218/4559d6fb-eca0-41fd-baca-a31aaa8408fb">

### What alternative solutions did you explore? (Optional)

**Comment by Sandeep7409** [comment]
First name and Last name allow max 50 characters and tooltip shows all 50 characters. I don't understand what is the meaning of "the end of the tooltip is broken".  Please include some actual and expected result in issue description to make it clear.

**Comment by alitoshmatov** [comment]
@Sandeep7409 I think they meant pointer of tooltip rather than its content

**Comment by priya-zha** [comment]
Yes, I meant the pointer of the tooltip. Sorry for the confusion.

**Comment by Sandeep7409** [comment]
I am not able to run code on my machine so haven't yet tried to solve it but my comment might be helpful If somebody else is already working on it. I feel like problem is with span tag used for display name. When display name is long, span width increase according to content. Binding tooltip to parent div of span tag might solve problem.

![image](https://github.com/Expensify/App/assets/9333046/309cf23c-0dbe-439d-9f9e-2541dbb6ce0e)

![image](https://github.com/Expensify/App/assets/9333046/cdfef0bc-4ed9-4c35-a90d-be397fce27ff)

**Comment by johncschuster** [comment]
@aimane-chnaif can you take a look at the above [proposal](https://github.com/Expensify/App/issues/21100#issuecomment-1614260482). Is this one sufficient?

**Comment by alitoshmatov** [comment]
@aimane-chnaif I think this one is expected, since the logic is the pointer pointing to the center of text

**Comment by Sandeep7409** [comment]
@alitoshmatov @aimane-chnaif More elegant solution is to use left:50% instead of calculated left css value. It does not even require change of moving Tooltip from span tag to parent div. 

![image](https://github.com/Expensify/App/assets/9333046/c5b15381-c450-45d4-9ba7-1cbe9abdfc04)

**Comment by Sandeep7409** [comment]
@alitoshmatov In that case @aimane-chnaif found the solution to the problem I think.

**Comment by cead22** [comment]
> First name and Last name allow max 50 characters and tooltip shows all 50 characters. I don't understand what is the meaning of "the end of the tooltip is broken". Please include some actual and expected result in issue description to make it clear.

@johncschuster @kbecciv I thought the same as this, can you please update the description to explain exactly what the issue is please?

Also, is there a difference between the two videos? If they're showing the same thing, can we remove one? And if the issue can be shown on a screenshot, can we include one as well?

**Comment by cead22** [comment]
> Web -Tooltip for user in group is offcenter and displayed over report

@alitoshmatov can you update your proposal to more clearly describe the issue
1. The tooltip isn't off center, it's the little triangle that's off center, right?
2. What do you mean displayed over report? Maybe I'm misunderstanding, but the screenshot in your solution makes me think your solution doesn't solve this

<img width="864" alt="image" src="https://github.com/Expensify/App/assets/165133/3489bac4-c1e4-4b81-a906-87b582400f20">

**Comment by alitoshmatov** [comment]
@cead22 Updated proposal.
1. You are right the pointer of tooltip is not pointing to the center of the text.
2. My mistake, I just copy-pasted the title of the issue at that time. I don't think it is an issue that tooltip is getting out of RHP.

**Comment by johncschuster** [comment]
Thanks for your comments about the description being unclear! I've updated it to say "pointer of the tooltip" to refer specifically to the triangle part of the tooltip. I've also clarified that, in this case, broken is referring to the pointer of the tooltip being not-centered.

**Comment by johncschuster** [comment]
@aimane-chnaif, earlier in the thread, you said @alitoshmatov's solution looks good. Do you still think that's the best path forward? If so, I'll get assignments going.

**Comment by johncschuster** [comment]
Thanks, @aimane-chnaif! Do you / @cead22 feel the problem statement is now clear after my adjustments? I'm happy to crystalize it further if not!

**Comment by cead22** [comment]
Reviewed, looks good

**Comment by alitoshmatov** [comment]
PR is ready - https://github.com/Expensify/App/pull/22869

**Comment by johncschuster** [comment]
@aimane-chnaif bump on the above checklist

**Comment by alitoshmatov** [comment]
Looks like we are ready to proceed 🙌

**Comment by alitoshmatov** [comment]
@spcheema I think it is not related to this issue and issue existed even before - https://github.com/Expensify/App/issues/23784#issuecomment-1655636942

**Comment by alitoshmatov** [comment]
@spcheema  https://github.com/Expensify/App/issues/23784 is created for this

**Comment by cead22** [comment]
Not overdue. Working on payments

**Comment by johncschuster** [comment]
_Are_ we working on payment? It seems like the behavior isnt' resolved yet, right?

**Comment by alitoshmatov** [comment]
@johncschuster Similar issue did come up, but it is not related to this and has a different root cause

**Comment by johncschuster** [comment]
@alitoshmatov can you share that similar issue here?

**Comment by alitoshmatov** [comment]
@johncschuster This one https://github.com/Expensify/App/issues/23784

**Comment by johncschuster** [comment]
Thanks for surfacing that, @alitoshmatov! I'll get payments issued now.

**Comment by johncschuster** [comment]
## Summary:

External issue reporter - @priya-zha = $250 (paid through Upwork)
Contributor that fixed the issue - @alitoshmatov - $1500 (paid through Upwork)
Contributor+ that helped on the issue and/or PR - @aimane-chnaif - $1500 (will request payment through NewDot)

**Comment by johncschuster** [comment]
Ah, thanks for calling that out! I'll get payment issued via Upwork for you then.

**Comment by johncschuster** [comment]
I have issued payment to @aimane-chnaif via Upwork.

@priya-zha, I've sent the invite to you via Upwork. Can you accept that so I can issue payment?

**Comment by priya-zha** [comment]
Thanks.. Accepted.

**Comment by johncschuster** [comment]
Thank you! I've issued payment to @priya-zha.

**Comment by cead22** [comment]
On hold for payment and not overdue

**Comment by johncschuster** [comment]
Uhh... I don't think that's right. I issued payment, and it looks like _I totally forgot to close the issue._ 🤦 

Closing!
